MovieGallery/Pages/Search.cshtml.cs

```diff
@@ -34,40 +36,66 @@ public void OnGet(string SearchbyYear)
                 ShowsJson = JsonConvert.DeserializeObject<Shows>(json);
             }
 
-            if (SearchbyYear == null)
-            {
-                InitialCheckList = true;
-                MoviesResult = MoviesJson;
-                ShowsResult = ShowsJson;
-            }
-            else
-            {
-                Query = SearchbyYear;
-                long inputYear = Convert.ToInt64(Query);
+            InitialCheckList = true;
+            MoviesResult = MoviesJson;
+            ShowsResult = ShowsJson;
 
-                var result = MoviesJson.Items.Where(x => x.Year == inputYear).ToList();
-                MoviesResult = new Movies()
+        }
+
+        public void OnPost(long? SearchbyYear)
```

Comment:
By OnPost method we can pass variables of type int,long,binary etc for variable "searchbyyear" as opposed to get where only string is passed. Since year is a long, we need not convert string to long. This also reduces the use variables "Query" and "inputYear" as we can directly use "searchbyyear" variable to get the result.
Also, we use long? here instead of long as we need to compare searchByYear with null. This gets rid of the warning that long cannot be nullable.

Source:
Difference between get and post method with advantages and disadvantages: https://www.guru99.com/difference-get-post-http.html

Use of nullable int, applied same logic to nulable long: https://stackoverflow.com/questions/6389437/explanation-of-int-vs-int

Cons of converting string to int : https://www.tutorialsteacher.com/articles/convert-string-to-int#:~:text=Cons%3A,null%20or%20invalid%20numeric%20string.

Reply:
Good.